cmd/cli/kubectl-kyverno/commands/test/output.go

```diff
@@ -1,30 +1,174 @@
 	policyreportv1alpha2 "github.com/kyverno/kyverno/api/policyreport/v1alpha2"
 	"github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/apis/v1alpha1"
 	"github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/output/color"
 	"github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/output/table"
 	engineapi "github.com/kyverno/kyverno/pkg/engine/api"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
+func printCheckResult(
+	out io.Writer,
+	checks []v1alpha1.CheckResult,
+	responses []engineapi.EngineResponse,
+	rc *resultCounts,
+	resultsTable *table.Table,
+) error {
+	testCount := 1
+	for _, check := range checks {
+		// filter engine responses
+		matchingEngineResponses := responses
+		// 1. by resource
+		if check.Match.Resource != nil {
+			var filtered []engineapi.EngineResponse
+			for _, response := range matchingEngineResponses {
+				errs, err := assert.Validate(context.Background(), check.Match.Resource.Value, response.Resource.UnstructuredContent(), nil)
+				if err != nil {
+					return err
+				}
+				if len(errs) == 0 {
+					filtered = append(filtered, response)
+				}
+			}
+			matchingEngineResponses = filtered
+		}
+		// 2. by policy
+		if check.Match.Policy != nil {
+			var filtered []engineapi.EngineResponse
+			for _, response := range matchingEngineResponses {
+				data, err := runtime.DefaultUnstructuredConverter.ToUnstructured(response.Policy().MetaObject())
+				if err != nil {
+					return err
+				}
+				errs, err := assert.Validate(context.Background(), check.Match.Policy.Value, data, nil)
+				if err != nil {
+					return err
+				}
+				if len(errs) == 0 {
+					filtered = append(filtered, response)
+				}
+			}
+			matchingEngineResponses = filtered
+		}
+		for _, response := range matchingEngineResponses {
+			// filter rule responses
+			matchingRuleResponses := response.PolicyResponse.Rules
+			if check.Match.Rule != nil {
+				var filtered []engineapi.RuleResponse
+				for _, response := range matchingRuleResponses {
+					data := map[string]any{
+						"name": response.Name(),
+					}
+					errs, err := assert.Validate(context.Background(), check.Match.Rule.Value, data, nil)
+					if err != nil {
+						return err
+					}
+					if len(errs) == 0 {
+						filtered = append(filtered, response)
+					}
+				}
+				matchingRuleResponses = filtered
+			}
+			for _, rule := range matchingRuleResponses {
+				// perform check
+				data := map[string]any{
+					"name":     rule.Name(),
+					"ruleType": rule.RuleType(),
+					"message":  rule.Message(),
+					"status":   string(rule.Status()),
+					// generatedResource unstructured.Unstructured
+					// patchedTarget *unstructured.Unstructured
+					// patchedTargetParentResourceGVR metav1.GroupVersionResource
+					// patchedTargetSubresourceName string
+					// podSecurityChecks contains pod security checks (only if this is a pod security rule)
+					"podSecurityChecks": rule.PodSecurityChecks(),
+					"exception ":        rule.Exception(),
```

Comment:
@eddycharly : Did you intentionally include a space in the "exception" field name or was that a typo?

```suggestion
					"exception":         rule.Exception(),
```

Reply:
Looks like a typo, can you confrim @eddycharly ?